### ISSUE TITLE
Error when reading mdocs with "...VS" entries of length 3

### ISSUE THREAD

**Issue by m-albert** [teamtomo/mdocfile#9]
Hi, I'm starting to work with SerialEM data and stumbled upon this great package :)

I wanted to report a problem I encountered: `mdocfile.read('example.mrc.mdoc')` complains about the following:

```
ValidationError: 3 validation errors for MdocSectionData
XedgeDxyVS
  wrong tuple length 3, expected 2 (type=value_error.tuple.length; actual_length=3; expected_length=2)
YedgeDxyVS
  value is not a valid tuple (type=type_error.tuple)
AlignedPieceCoordsVS
  wrong tuple length 3, expected 2 (type=value_error.tuple.length; actual_length=3; expected_length=2)
```
Indeed the entries for "XedgeDxyVS" and "AlignedPieceCoordsVS" in my file are of length 3. While on one side I'm trying to understand how to interpret these three values, I'd guess `mdocfile` would want to read them (since my mdoc should be a valid SerialEM output).

Cheers!

**Comment by m-albert** [comment]
As a follow up, if I modify the types in `section_data` as such:

`    XedgeDxyVS: Optional[Tuple[float, float, float]]`

I'm getting a different error instead:

```
ValidationError: 1 validation error for MdocSectionData
YedgeDxyVS
  value is not a valid tuple (type=type_error.tuple)
```

**Comment by alisterburt** [comment]
Thanks for the kind words and the report @m-albert ! I'm sorry the package wasn't quite working out of the box with your data, hopefully we can get you up and running quickly - I'll take a look tomorrow

Could you share your file so I can use it for debugging? Ideally as a GitHub gist or similar 🙂

**Comment by m-albert** [comment]
Thanks for your quick response @alisterburt :)

[Here's](https://gist.github.com/m-albert/3755e09511f717bf6b15efe0226166be) the file as a gist.

**Comment by alisterburt** [comment]
glad to help! Thanks for opening the issue, issues like this have probably come up before but until we know about them it's tough to do much about it

merged #10 with a fix and v0.0.8 should be released shortly, progress on deployment can be seen at
https://github.com/teamtomo/mdocfile/actions/runs/4628299599

**Comment by alisterburt** [comment]
just fyi, your mdocs look a little different to ones I've seen before in that they have two types of section data `ZValue` and `MontSection` -> these contain different data so now a few columns will contain some NaN values

**Comment by m-albert** [comment]
> merged https://github.com/teamtomo/mdocfile/pull/10 with a fix and v0.0.8 should be released shortly, progress on deployment can be seen at
https://github.com/teamtomo/mdocfile/actions/runs/4628299599

Thanks so much for this extremely quick fix :) I'll start using it right away from the main branch.

> glad to help! Thanks for opening the issue, issues like this have probably come up before but until we know about them it's tough to do much about it

You're welcome, and thanks for sparing me and others redundant efforts!

> just fyi, your mdocs look a little different to ones I've seen before in that they have two types of section data ZValue and MontSection -> these contain different data so now a few columns will contain some NaN values

Exactly, the last entry type describes the montage of everything above (and I'd guess would not go to values higher than 0).  You even added a new column to the output dataframe :)
